app/models/report.rb

```diff
@@ -4,14 +4,37 @@ class Report < ApplicationRecord
   belongs_to :user
   has_many :comments, as: :commentable, dependent: :destroy
 
+  has_many :mentioned_by_reports, class_name: 'ReportLink', foreign_key: 'mentioned_report_id', dependent: :destroy, inverse_of: :mentioned_report
+  has_many :mentioning_reports, through: :mentioned_by_reports, source: :mentioning_report
+  has_many :mentions_to_reports, class_name: 'ReportLink', foreign_key: 'mentioning_report_id', dependent: :destroy, inverse_of: :mentioning_report
+  has_many :mentioned_reports, through: :mentions_to_reports, source: :mentioned_report
+
   validates :title, presence: true
   validates :content, presence: true
 
+  def create_report_link
+    mentioned_by_reports.destroy_all
+    mentioning_report_ids = content.scan(%r{http://127.0.0.1:3000/reports/(\d+)})
+
+    mentioning_report_ids.each do |id|
+      mentioning_report = Report.find(id[0].to_i)
```

Comment:
```suggestion
      mentioning_report = Report.find(id[0])
```

（暗黙の型変換がよいかどうかの議論はさておき）to_iなしでも動きそうです

Reply:
動きました！

キャストする必要がある場合はなぜそれをしたのかわかるようなコメントや適切な変数名で再定義することで安全に運用出来ますか？

Reply:
> キャストする必要がある場合はなぜそれをしたのかわかるようなコメントや適切な変数名で再定義することで安全に運用出来ますか？

単純な型変換（文字列←→シンボル、文字列←→数値）みたいなものであれば、「ここは型を揃えないと動かないんだろうな〜」ということが想像できるのでわざわざコメントを書いたり、変数に入れ替えたりすることはあまりないかもしれません。

何か不自然な型変換（第三者が疑問を覚えるようなケース）であればコメント等を入れた方がいいかもしれません。